### PR TITLE
feat: shim yarn commands

### LINKED ISSUE #136: Automatically use .yvmrc upon entering a directory
**Is your feature request related to a problem? Please describe.**
When working with multiple projects that utilize different versions of yarn, I always have to type `yvm use` in order to have yvm set the correct version of yarn that my project has. 

**Describe the solution you'd like**
When I switch into a directory that already has a `.yvmrc` file, I would like yvm to automatically run `yvm use` so that I don't have to do it manually.

### PR DESCRIPTION
**Pull request by aeldamaty**
## Description

Should resolve #136. All yarn commands will now be shimmed. YVM will look for a `.yvmrc` file to see which version of yarn should be used. If none are found, it will used the last globally set default version of yarn.

### DevQA Prep

- Run `make install`
- Open a new terminal session. `echo $PATH`. You should see the shim directory at the front of your path.

### DevQA Steps
- Run any yarn command. If you do not have a default yarn version set, this will be executed using the `default` `stable` version.
- Switch into any folder with a .yvmrc file with a yarn version that is not installed. Run a yarn command. This will download and install the yarn version before running the command.

### CODE REVIEW

**Comment by mcataford** [comment]
@aeldamaty QA steps unclear, can you rephrase/rewrite and run through them?

File: src/shim/yarn [hunk 1]
```diff
@@ -0,0 +1,40 @@
+set -e
```
Comment (jakebolam) [inline src/shim/yarn]: Hash bang?

**Comment by jakebolam** [comment]
@aeldamaty what is the motivation behind this?

You'll notice there's some been some PR's to assist in this effort:
https://github.com/tophat/yvm/pull/118

**Comment by jakebolam** [comment]
Have put up #189 which removes the need for this

**Comment by aeldamaty** [comment]
@jakebolam so the motivation behind this is to get rid of the need to type in `yvm use`. It shims any yarn command and automatically appends the correct version to the PATH. 

It will probably need some refactoring once your global version of yarn PR goes in, but essentially it would work as follows. 

- Any yarn command you entered would be shimmed
- It would determine the version of yarn to use by looking for a .yvmrc file
- If no .yvmrc file exists, it will use the global version
- If none set it would error out. 

Also, the way it currently works, if the user ever types in `yvm use` they have essentially taken over control for just this session and it will use the version of yarn they specify at all times without shimming.

File: src/shim/yarn [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+program="${0##*/}"
```
Comment (jakebolam) [inline src/shim/yarn]: Can we get tests for his file in CI somehow

File: src/shim/yarn [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+if [ $current_path = *.yvm/versions/v* ]; then
+    yarn_version="$(yarn --version)"
+    yarn_version_base_directory="${yvm_dir}/versions/v${yarn_version}/"
+    yarn_directory="${yarn_version_base_directory}bin/"
+    exec -a "$yarn_directory" "$program" "$@"
+    return
+fi
+
+if [ "$program" = "yarn" ]; then
+    yarn_version="${LAST_USED_YARN_VERSION:-}"
```
Comment (jakebolam) [inline src/shim/yarn]: Can this logic be moved to js?

**Comment by jakebolam** [comment]
@aeldamaty I like the motivation to solve the cd'ing into directories etc by shimming the yarn command. Once the global version PR is in, lets totally figure out this logic

**Comment by mcataford** [comment]
@aeldamaty Let's revive and drive this one home 🚀

**Comment by mcataford** [comment]
@aeldamaty Any news on this? :rocket:

File: src/yvm.fish [hunk 4]
```diff
@@ -15,7 +15,7 @@ function yvm
         if [ -z "$NEW_FISH_USER_PATHS" ]
             yvm_err "Could not get new path from yvm"
         else
-            set -U fish_user_paths $NEW_FISH_USER_PATHS
+            set -U fish_user_paths (string split ' ' -- $NEW_FISH_USER_PATHS)
```
Comment (iamogbz) [inline src/yvm.fish]: This is to fix fish not splitting on spaces but [on new lines](https://fishshell.com/docs/current/faq.html)
